copier/main.py

```diff
@@ -767,6 +767,41 @@ def run_update(self) -> None:
             )
         self._apply_update()
 
+    def run_check(self) -> None:
+        """Check if a subproject is using the latest version of its template.
+
+        See [checking a project][checking-a-project].
+        """
+        # Check all you need is there
+        if self.subproject.template is None or self.subproject.template.ref is None:
+            raise UserMessageError(
+                "Cannot check because cannot obtain old template references "
+                f"from `{self.subproject.answers_relpath}`."
+            )
+        if self.template.commit is None:
+            raise UserMessageError(
+                "Checking is only supported in git-tracked templates."
+            )
+        if not self.subproject.template.version:
+            raise UserMessageError(
+                "Cannot check: version from last update not detected."
+            )
+        if not self.template.version:
+            raise UserMessageError("Cannot check: version from template not detected.")
+
+        if self.template.version > self.subproject.template.version:
+            if not self.quiet:
+                # TODO Unify printing tools
+                print(
+                    f"NEW template version available. "
+                    f"Currently using {self.subproject.template.version}, "
+                    f"latest is {self.template.version}.",
+                    file=sys.stderr)
+            raise SubprojectOutdatedError()
+        elif not self.quiet:
+            # TODO Unify printing tools
+            print("Project is up-to-date 🎉", file=sys.stderr)
```

Comment:
I'd like it better if the exception contained a couple of attributes to extract the current and latest versions from it.

Then I'd like it better too if the printing is done only in the CLI part.

Finally, I don't think we need to print anything when all is OK.